### PR TITLE
SN-5612 cltool debug

### PR DESCRIPTION
**Pull request by waltjohnson**
Add runtime and count for tx packet, rx packet, rx byte to cltool.

### CODE REVIEW

File: cltool/setup_debug_acu.sh
Comment (kylemallory) [inline cltool/setup_debug_acu.sh:1]: This shouldn't be in the public Repo, since its customer-specific

File: cltool/src/cltool_main.cpp [hunk 1]
```diff
@@ -264,7 +264,7 @@ static void cltool_dataCallback(InertialSense* i, p_data_t* data, int pHandle)
         cout.flush();
     else if (!g_inertialSenseDisplay.ExitProgram()) // don't process any additional data once we've been told to exit
         g_inertialSenseDisplay.ProcessData(data);
-}
+    }
```
Comment (kylemallory) [inline cltool/src/cltool_main.cpp:267]: Errant whitespace?

File: cltool/src/cltool_main.cpp [hunk 2]
```diff
@@ -714,6 +714,12 @@ static int cltool_dataStreaming()
         cout << "Failed to open serial port at " << g_commandLineOptions.comPort.c_str() << endl;
         return -1;    // Failed to open serial port
     }
+    g_inertialSenseDisplay.SetSerialPort(inertialSenseInterface.SerialPort());
```
Comment (kylemallory) [inline cltool/src/cltool_main.cpp:717]: This seems like a strange place (`cltool_requestDataSets()`) to be doing ISDisplay setup.  Should be be moved into `cltool_streamData()` or `cltool_setupCommunications()`?

File: src/ISDataMappings.cpp [hunk 3]
```diff
@@ -2975,6 +2975,14 @@ uint32_t cISDataMappings::DefaultPeriodMultiple(uint32_t dataId)
     case DID_INL2_STATES:
     case DID_ROS_COVARIANCE_POSE_TWIST:
     case DID_INL2_MAG_OBS_INFO:
+    case DID_GPX_DEV_INFO:
```
Comment (kylemallory) [inline src/ISDataMappings.cpp:2978]: Nice!

File: src/ISDisplay.cpp [hunk 4]
```diff
@@ -225,19 +225,41 @@ void cInertialSenseDisplay::GoToColumnAndRow(int x, int y)
...
+	{
+		m_startMs = current_timeMs();
+	}
+
+	// cltool runtime
+	double runtime = 0.001 * (current_timeMs() - m_startMs);
+
+	std::ostringstream stream;
+
+	stream << "$ Inertial Sense.  Connected.  CTRL-C to terminate.  ";
```
Comment (kylemallory) [inline src/ISDisplay.cpp:243]: Its not a big deal, and it was this way before your work, but we should try and consolidate these messages/string.  Would a `generateHeader(bool connected)` make more sense, and then have the resulting string be a concatenation of these fields, if we're connected (or not)?

File: src/ISDisplay.h [hunk 5]
```diff
@@ -138,6 +139,8 @@ class cInertialSenseDisplay
 	bool UploadNeeded() { bool uploadNeeded = m_editData.uploadNeeded; m_editData.uploadNeeded = false; return uploadNeeded; };
 	edit_data_t *EditData() { return &m_editData; }
 	void setOutputOnceDid(int did) { m_outputOnceDid = did; m_interactiveMode = m_outputOnceDid == 0; }
+	void SetSerialPort(serial_port_t* port) { m_port = port; }
```
Comment (kylemallory) [inline src/ISDisplay.h:142]: How does this work when connected to multiple devices (ie `-c *`)?  How does normal/current display stuff work in that situation?

**Review by kylemallory — CHANGES_REQUESTED** [review]

**Review by tmcd249 — APPROVED** [review]

File: cltool/src/cltool_main.cpp (hunk 2 above)
Comment (waltjohnson) [reply cltool/src/cltool_main.cpp:717]: Moved into cltool_setupCommunications()

File: src/ISDisplay.cpp (hunk 4 above)
Comment (waltjohnson) [reply src/ISDisplay.cpp:243]: Done

File: src/ISDisplay.h (hunk 5 above)
Comment (waltjohnson) [reply src/ISDisplay.h:142]: It only shows stats for the first device.

**Pull request merged**